app.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book_manager'
+require_relative 'person_manager'
+require_relative 'rental_manager'
+class App
+  def initialize
+    @book_manager = BookManager.new
+    @person_manager = PersonManager.new
+    @rental_manager = RentalManager.new
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each_with_index { |book, index| puts "#{index}) #{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each do |person|
+      if person.is_a?(Teacher)
+        puts "#{person.name} (Teacher, Age: #{person.age}, Specialization: #{person.specialization})"
+      else
+        puts "#{person.name} (Student)"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Enter person details:'
+    name = get_user_input('Name: ')
+
+    puts 'Select person type:'
+    puts '1. Student'
+    puts '2. Teacher'
+    choice = get_user_input('Enter the option (1/2): ').to_i
+
+    case choice
+    when 1
+      @person_manager.create_student(name)
+    when 2
+      create_teacher(name)
+    else
+      puts 'Invalid choice'
+    end
+  end
+
+  def create_teacher(name)
+    age = get_user_input('Age: ').to_i
+    specialization = get_user_input('Specialization: ')
+    @person_manager.create_teacher(name, age, specialization)
+  end
+
+  def create_book
+    puts 'Enter book details:'
+    title = get_user_input('Title: ')
+    author = get_user_input('Author: ')
+
+    @book_manager.create_book(title, author)
+  end
+
+  def create_rental
+    puts 'Enter rental details:'
+    book_number = select_book
+    person_number = select_person
+    date = get_user_input('Rental date (yyyy/mm/dd): ')
+
+    if valid_selection?(book_number, @book_manager.books.length) &&
+       valid_selection?(person_number, @person_manager.people.length)
+
+      book = @book_manager.books[book_number]
+      person = @person_manager.people[person_number]
+      @rental_manager.create_rental(book, person, date)
+    else
+      puts 'Invalid book or person selection.'
+    end
```

Comment:
- [ ] Hey there! 👋 Your create_rental method definition is pretty good, but there's room for some improvements in terms of user experience. Currently, when you list both books and people to choose from, they're not numbered, and this can be a bit confusing for users. 📚👥 To enhance the user experience, it would be beneficial to modify all related methods used in the create_rental process.
- [ ] What we can do is add numbers. This makes it easier for users to select their choices without any ambiguity. 🤔🔢 By doing this, you'll significantly improve the user experience in your app and make it more user-friendly. 🌟 Feel free to make these adjustments to ensure a smoother and more intuitive interaction for your users! 🛠️😊